src/web/src/pages/CourseExplorer.vue

```diff
@@ -125,6 +145,13 @@ export default {
     };
   },
   async created() {
+    if (this.selectedSemester === "") {
+      const querySemester = this.$route.query.semester;
+      this.selectedSemester =
+        querySemester && querySemester != "null"
```

Comment:
> So which issue is this PR for? The title seems to indicate #191 but looks like #192 is also in here

Yes, Ike and I are working on the same branch for #191 and #192 as we discussed before with Jacob. We thought this would make both of us easier.